### ISSUE TITLE
Font contrast in documentation editor is low

### ISSUE THREAD

**Issue by pkcakeout** [comic/grand-challenge.org#3732]
I am editing a help-page on grand-challenge right now, and found that the contrast of the editor is very low:

![image](https://github.com/user-attachments/assets/004f053f-b437-4e3c-bcc5-fb981ecf033c)

How to reproduce: 

1. Open https://grand-challenge.org/documentation/create/
2. (not sure if these next steps are required) Type some Markdown
3. Switch to preview and back to Editor

Problem: The editor font is gray. I think black would be the logical choice for easy reading and max contrast? Like here on GitHub!

**Comment by jmsmkn** [comment]
I immediately thought that this was coming from the theme but it seems not the case: https://bootswatch.com/flatly/

**Comment by pkcakeout** [comment]
Inspect element reveals it comes from the class `.form-control`:

![image](https://github.com/user-attachments/assets/ec216d57-3058-4a5b-adc7-b58f10443218)

I do not know what that means though. It seems that the other fonts are not even black either:

![image](https://github.com/user-attachments/assets/16b3bd92-c252-4211-9574-ee796dcf6f3b)

Very interesting. Was this hand-picked somehow? Here a contrast comparison of `#000000` vs `#212529`, upper part with  a sharp edge the lower parth with a blurred edge:

![image](https://github.com/user-attachments/assets/56fb83f4-c99d-4ebd-9616-1f01216df605)

I think `#212529` is fine as well, just interesting.

**Comment by koopmant** [comment]
This is in line with flatly on bootstrap 4. For a theme preview on bootstrap 4, see https://bootswatch.com/4/flatly/

If we want to change this, we should upgrade. Assuming we don't want to, I'd say we can close this?

**Comment by pkcakeout** [comment]
> Assuming we don't want to, I'd say we can close this?

I personally, find it very unnecessary to have low contrast in a text-editor that is there for writing fairly complex documents. The design there should, imo, optimize the productivity of the user (which is mainly us for that page) and I would much prefer black/white contrast for that.

Anyway, my 2-cents. Can be done "someday". I have written a little Tampermonkey-script on my laptop that sets the style for the editor to what I like :d

**Comment by pkcakeout** [comment]
No... cannot turn it off. Seems to require a recompile. But in general I am inferring the following:

This might only be a problem on "standard res" screens (no 4k) in the first place with gray-value smoothing which smears out the fonts and therefore reduces the sharp contrast at the edges of fonts. This is much less of an issue on high-density screens where you would have many pixels to render tiny text, but on 1080p screens there are fewer pixels for this and subpixel smoothing using grayscale instead of the rgba-style causes the contrast to drop.

<img width="722" height="433" alt="Image" src="https://github.com/user-attachments/assets/a5a4b5ea-5683-4c46-abcb-8f175cbb9072" />

This is how "black" text renders for me in firefox as a reference. Also not 100% black. I am not 100% sure how I can force BGR sub pixel smoothing on my laptop, but that seems to be the main issue. I am sure that it would be better then. When go to "4k" on my screen I get this as subpixel smoothing:

<img width="504" height="319" alt="Image" src="https://github.com/user-attachments/assets/7fb0afa7-7eee-471d-a0a6-6c5c8be039fd" />

Which has a lot more absolute contrast. However, my intel graphics card struggles with rendering some applications in 4k which is why I do not like working in 4k on my laptop.

----------------

# TL;DR

I think the issue is observable on most 1080p screens, actually, when the fonts are set to be relatively tiny. Most browsers implement grayscale antialiasing nowadays it seems while BGR/RGB smoothing would provide better contrast.

**Comment by pkcakeout** [comment]
Yes, indeed. For me it is "Liberation Mono". "SFMono-Regular" should be the one you are seeing from the CSS-style and documentation, right? Anyways, ehm... I do not know what to do anymore.

I think a darker color would still be the simplest solution to maximize contrast? On the screen I am working on right now, I do not experience a readability issue, but that is likely only because I selected this panel because it had very awesome contrast.
